bazel/testing/lit_autoupdate_base.py

```diff
@@ -190,35 +203,61 @@ class CheckLine(Line):
 
     def __init__(
         self,
+        test: str,
         out_line: str,
-        line_number_format: str,
+        line_number_delta_prefix: str,
         line_number_pattern: Pattern,
     ) -> None:
         super().__init__()
+        self.filename = Path(test).name
         self.indent = ""
         self.out_line = out_line.rstrip()
-        self.line_number_format = line_number_format
+        self.line_number_delta_prefix = line_number_delta_prefix
         self.line_number_pattern = line_number_pattern
-        self.line_numbers = [
-            int(n) - 1 for n in line_number_pattern.findall(self.out_line)
-        ]
+
+        # If any match has a match specific to this file, use the first line for
```

Comment:
Should this be:
```suggestion
        # If any match is specific to this file, use the first line for
```
? I'm not sure what it means for a match to have a match. Or maybe
```suggestion
        # If any match refers to a line in this file, use the first line for
```
I also misinterpreted "first line" as meaning "the first line in the file", not "the first line in the match", maybe "first such line" would be clearer?